diff -urN --exclude='*~' --exclude='.#*' --exclude=CVS --exclude=RCS --exclude='*,v' wmgtemp-0.5/src/wmgtemp.c wmgtemp-0.5-new/src/wmgtemp.c
--- wmgtemp-0.5/src/wmgtemp.c		2001/12/18 16:58:07	1.1
+++ wmgtemp-0.5-new/src/wmgtemp.c	2001/12/18 17:01:48	1.2
@@ -56,6 +56,7 @@
 #define W83627HF 2
 #define AS99127F 3
 #define ADM1021  4
+#define IT87     5 /* ITE IT87xx chips */
 
 #define OPT_STRING	"g:s:hH:w:m:M:a:e:u:t"
 
@@ -190,6 +191,14 @@
       SENSOR_DEF_SYS = SENSORS_W83781D_TEMP1;
       chiptype = W83627HF;
     }
+    else if(!strcmp(name->prefix, SENSORS_IT87_PREFIX)) {
+      chipname = SENSORS_IT87_PREFIX;
+      BitOn(SENSOR_DISP, CPU);
+      BitOn(SENSOR_DISP, SYS);
+      SENSOR_DEF_CPU = SENSORS_IT87_TEMP2;
+      SENSOR_DEF_SYS = SENSORS_IT87_TEMP1;
+      chiptype = IT87;
+    }
     
     if(chiptype == -1) 
       name = sensors_get_detected_chips(&chip_nr);
