diff -urN --exclude='*~' --exclude='.#*' --exclude=CVS --exclude=RCS --exclude='*,v' wmgtemp-0.5/src/wmgtemp.c wmgtemp-0.5-new/src/wmgtemp.c
--- wmgtemp-0.5/src/wmgtemp.c		2001/12/18 17:43:13	1.5
+++ wmgtemp-0.5-new/src/wmgtemp.c	2001/12/18 19:32:26	1.6
@@ -57,6 +57,7 @@
 #define AS99127F 3
 #define ADM1021  4
 #define IT87     5 /* ITE IT87xx chips */
+#define DETECT 255 /* other chip with temp1 and temp2 features */ 
 
 #define OPT_STRING	"g:s:hH:w:m:M:a:e:u:t"
 
@@ -123,6 +124,11 @@
   int i = 0;
   short tmp_swap;
 
+  const sensors_feature_data *feature = NULL;
+  int temp1_number, temp2_number;
+  int feature_nr1, feature_nr2;
+
+
   BitOn(SENSOR_DISP, WARN_NONE);
   BitOn(SENSOR_DISP, TSCALE_CELSIUS);
   BitOn(SENSOR_DISP, GRAPH_LINE);
@@ -199,7 +205,35 @@
       SENSOR_DEF_SYS = SENSORS_IT87_TEMP1;
       chiptype = IT87;
     }
-    
+    else { /* check features */
+      temp1_number = temp2_number = -1;
+      feature_nr1  = feature_nr2  = 0;
+
+      /* while there are any features in current chip  */
+      /* and we haven't found both "temp1" and "temp2" */
+      while((feature=sensors_get_all_features(*name, &feature_nr1, &feature_nr2)) != NULL &&
+	    (temp1_number < 0 || temp2_number < 0)) {
+	if(strcmp(feature->name, "temp1") == 0 &&
+	   sensors_get_ignored(*name, feature->number))
+	  temp1_number = feature->number;
+	if(strcmp(feature->name, "temp2") == 0 &&
+	   sensors_get_ignored(*name, feature->number))
+	  temp2_number = feature->number;
+      }
+
+      /* if both "temp1" and "temp2" were fount, feature is not NULL */
+      /* and both temp1_number and temp2_number are positive I think */
+      /* if one feels paranoid, he/she can add two booleans flags    */
+      if(feature != NULL) {
+	chipname = name->prefix;
+	BitOn(SENSOR_DISP, CPU);
+	BitOn(SENSOR_DISP, SYS);
+	SENSOR_DEF_CPU = temp2_number;
+	SENSOR_DEF_SYS = temp1_number;
+	chiptype = DETECT;
+      }
+    }
+
     if(chiptype == -1) 
       name = sensors_get_detected_chips(&chip_nr);
   }
